sclang compilation error in Linux

As before, I'm running Ubuntu Studio 14.04.3 LTS with gcc 4.8.4-2ubuntu1~14.04 and boost 1.55. Latest master fails with the following error:

    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp: In function ‘size_t fillClassRow(const PyrClass*, PyrMethod**, boost::executors::basic_thread_pool&)’:
    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:1388:62: error: no matching function for call to ‘irange(long unsigned int, size_t&)’
         for( int subClassIndex : boost::irange(0UL, numSubclasses) )
                                                                  ^
    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:1388:62: note: candidates are:
    In file included from /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:43:0:
    /home/nathan/src/supercollider/external_libraries/boost/boost/range/irange.hpp:208:5: note: template<class Integer> boost::integer_range<Integer> boost::irange(Integer, Integer)
         irange(Integer first, Integer last)
         ^
    /home/nathan/src/supercollider/external_libraries/boost/boost/range/irange.hpp:208:5: note:   template argument deduction/substitution failed:
    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:1388:62: note:   deduced conflicting types for parameter ‘Integer’ (‘long unsigned int’ and ‘unsigned int’)
         for( int subClassIndex : boost::irange(0UL, numSubclasses) )
                                                                  ^
    In file included from /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:43:0:
    /home/nathan/src/supercollider/external_libraries/boost/boost/range/irange.hpp:216:9: note: template<class Integer, class StepSize> boost::strided_integer_range<Integer> boost::irange(Integer, Integer, StepSize)
             irange(Integer first, Integer last, StepSize step_size)
             ^
    /home/nathan/src/supercollider/external_libraries/boost/boost/range/irange.hpp:216:9: note:   template argument deduction/substitution failed:
    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:1388:62: note:   deduced conflicting types for parameter ‘Integer’ (‘long unsigned int’ and ‘unsigned int’)
         for( int subClassIndex : boost::irange(0UL, numSubclasses) )
                                                                  ^
    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:1394:62: error: no matching function for call to ‘irange(long unsigned int, size_t&)’
         for( int subClassIndex : boost::irange(1UL, numSubclasses) ) {
                                                                  ^
    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:1394:62: note: candidates are:
    In file included from /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:43:0:
    /home/nathan/src/supercollider/external_libraries/boost/boost/range/irange.hpp:208:5: note: template<class Integer> boost::integer_range<Integer> boost::irange(Integer, Integer)
         irange(Integer first, Integer last)
         ^
    /home/nathan/src/supercollider/external_libraries/boost/boost/range/irange.hpp:208:5: note:   template argument deduction/substitution failed:
    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:1394:62: note:   deduced conflicting types for parameter ‘Integer’ (‘long unsigned int’ and ‘unsigned int’)
         for( int subClassIndex : boost::irange(1UL, numSubclasses) ) {
                                                                  ^
    In file included from /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:43:0:
    /home/nathan/src/supercollider/external_libraries/boost/boost/range/irange.hpp:216:9: note: template<class Integer, class StepSize> boost::strided_integer_range<Integer> boost::irange(Integer, Integer, StepSize)
             irange(Integer first, Integer last, StepSize step_size)
             ^
    /home/nathan/src/supercollider/external_libraries/boost/boost/range/irange.hpp:216:9: note:   template argument deduction/substitution failed:
    /home/nathan/src/supercollider/lang/LangSource/PyrObject.cpp:1394:62: note:   deduced conflicting types for parameter ‘Integer’ (‘long unsigned int’ and ‘unsigned int’)
         for( int subClassIndex : boost::irange(1UL, numSubclasses) ) {
                                                                  ^
    make[2]: *** [lang/CMakeFiles/libsclang.dir/LangSource/PyrObject.cpp.o] Error 1
    make[1]: *** [lang/CMakeFiles/libsclang.dir/all] Error 2
    make: *** [all] Error 2

Is my Boost version too old, maybe?

--- a/lang/LangSource/PyrObject.cpp
+++ b/lang/LangSource/PyrObject.cpp
@@ -1380,18 +1380,18 @@ static size_t fillClassRow(const PyrClass *classobj, PyrMethod** bigTable, boost
 
 	if (IsObj(&classobj->subclasses)) {
 		const PyrObject * subclasses = slotRawObject(&classobj->subclasses);
-		size_t numSubclasses = subclasses->size;
+		int numSubclasses = subclasses->size;
 
 		if( numSubclasses ) {
 
 			if( numSubclasses <= 2 ) {
-				for( int subClassIndex : boost::irange(0UL, numSubclasses) )
+				for( int subClassIndex : boost::irange(0, numSubclasses) )
 					result += fillClassRow( slotRawClass(&subclasses->slots[subClassIndex]), bigTable, pool );
 			} else {
 				typedef std::vector< boost::future<size_t> > VectorOfFutures;
 
 				VectorOfFutures subclassResults;
-				for( int subClassIndex : boost::irange(1UL, numSubclasses) ) {
+				for( int subClassIndex : boost::irange(1, numSubclasses) ) {
 					auto subclassResult = boost::async( pool, fillClassRow, slotRawClass(&subclasses->slots[subClassIndex]), bigTable, boost::ref(pool) );
 					subclassResults.emplace_back( std::move( subclassResult ) );
 				}

